lib/cloud_controller/dea/app_stager_task.rb

```diff
@@ -142,15 +165,19 @@ def ensure_staging_is_current!
           raise CloudController::Errors::ApiError.new_from_details('StagingError', "failed to stage application: can't retrieve staging status")
         end
 
-        if @app.staging_task_id != task_id
-          raise CloudController::Errors::ApiError.new_from_details('StagingError', 'failed to stage application: another staging request was initiated')
-        end
+        check_task_id unless use_http
```

Comment:
Because if we're using http, we check the task id when we receive the message (in the controller), and if it doesn't match, we fail early (discard the message).  We know it makes the paths slightly different for NATS and http, but we didn't think it made sense to get this far on the http path before discarding a no-longer-current staging response.